src/position-manager/CreditDelegationPositionManager.sol

```diff
@@ -0,0 +1,101 @@
+// SPDX-License-Identifier: UNLICENSED
+// Copyright (c) 2025 Aave Labs
+pragma solidity 0.8.28;
+
+import {SignatureChecker} from 'src/dependencies/openzeppelin/SignatureChecker.sol';
+import {SafeERC20, IERC20} from 'src/dependencies/openzeppelin/SafeERC20.sol';
+import {EIP712} from 'src/dependencies/solady/EIP712.sol';
+import {MathUtils} from 'src/libraries/math/MathUtils.sol';
+import {NoncesKeyed} from 'src/utils/NoncesKeyed.sol';
+import {EIP712Hash, EIP712Types} from 'src/position-manager/libraries/EIP712Hash.sol';
+import {ISpoke} from 'src/spoke/interfaces/ISpoke.sol';
+import {PositionManagerBase} from 'src/position-manager/PositionManagerBase.sol';
+import {ICreditDelegationPositionManager} from 'src/position-manager/interfaces/ICreditDelegationPositionManager.sol';
+
+/// @title CreditDelegationPositionManager
+/// @author Aave Labs
+/// @notice Position manager to handle credit delegation and borrow actions on behalf of users.
+contract CreditDelegationPositionManager is
+  ICreditDelegationPositionManager,
+  PositionManagerBase,
+  NoncesKeyed,
+  EIP712
+{
+  using SafeERC20 for IERC20;
+  using EIP712Hash for *;
+  using MathUtils for uint256;
+
+  /// @notice Mapping of credit delegations.
+  mapping(address owner => mapping(address spender => mapping(uint256 reserveId => uint256 amount)))
+    private _creditDelegations;
+
+  /// @dev Constructor.
+  /// @param spoke_ The address of the spoke contract.
+  constructor(address spoke_) PositionManagerBase(spoke_) {}
+
+  /// @inheritdoc ICreditDelegationPositionManager
+  function approveCreditDelegation(address spender, uint256 reserveId, uint256 amount) external {
+    _creditDelegations[msg.sender][spender][reserveId] = amount;
+    emit CreditDelegation(msg.sender, spender, reserveId, amount);
+  }
+
+  /// @inheritdoc ICreditDelegationPositionManager
+  function approveCreditDelegationWithSig(
```

Comment:
should we have something like `PositionManagerWithAllowanceBase.sol`, which implements the map and the 2 functions for approvals (with and without sig)? (to avoid duplication)

wdyt?

Reply:
ofc, still unique typehash, but we can pass typehash to the base